### PR TITLE
chore: add error handling 

### PR DESCRIPTION
**Pull request by TropicalDog17**
## Description

Add missing error handling and proper logging.

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling for deposit file operations with detailed error messages, enhancing error traceability and user feedback.
- **New Features**
	- Introduced specific error messages for file operations and message marshaling in the genesis package.
- **Refactor**
	- Enhanced the functionality of the `Set` method in the storage module to handle file closing errors more robustly by logging any encountered errors.

### CODE REVIEW

File: mod/cli/pkg/commands/genesis/deposit.go [hunk 1]
```diff
@@ -172,19 +172,29 @@ func writeDepositToFile(
 		0o644, //nolint:mnd // file permissions.
 	)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to open file: %w", err)
```
Comment (nidhi-singh02) [inline mod/cli/pkg/commands/genesis/deposit.go:175]: Can we use errors package to be consistent. Similar comment at other places.

**Comment by TropicalDog17** [comment]
Hi @itsdevbear @nidhi-singh02 please take a look

File: mod/cli/pkg/commands/genesis/deposit.go [hunk 2]
```diff
@@ -172,19 +172,29 @@
...
+			if err == nil {
+				err = errors.Newf("failed to close file: %w", cerr)
+			}
+		}
+	}()
 
 	bz, err := json.Marshal(depositMessage)
 	if err != nil {
-		return err
+		return errors.Newf("failed to marshal deposit message: %w", err)
```
[the remark this answers is not in the document]
Comment (nidhi-singh02) [reply mod/cli/pkg/commands/genesis/deposit.go:189]: I agree with coderabbitai here.

**Comment by TropicalDog17** [comment]
hi @itsdevbear please review

**Pull request closed without merging**